assets/js/components/product-control/index.js

```diff
@@ -1,54 +1,239 @@
+	return name.replace( re, '<strong>$&</strong>' );
+}
+
+const getInteractionIcon = ( isSelected = false ) => {
+	return isSelected ? <IconRadioSelected /> : <IconRadioUnselected />;
+};
 
 class ProductControl extends Component {
 	constructor() {
 		super( ...arguments );
 		this.state = {
-			list: [],
+			products: [],
+			product: 0,
+			variationsList: {},
+			variationsLoading: false,
 			loading: true,
 		};
 
 		this.debouncedOnSearch = debounce( this.onSearch.bind( this ), 400 );
+		this.debouncedGetVariations = debounce( this.getVariations.bind( this ), 200 );
+		this.renderItem = this.renderItem.bind( this );
+		this.onProductSelect = this.onProductSelect.bind( this );
 	}
 
 	componentDidMount() {
 		const { selected } = this.props;
 
 		getProducts( { selected } )
-			.then( ( list ) => {
-				this.setState( { list, loading: false } );
+			.then( ( products ) => {
+				products = products.map( ( product ) => {
+					const count = product.variations ? product.variations.length : 0;
+					return {
+						...product,
+						parent: 0,
+						count: count,
+					};
+				} );
+				this.setState( { products, loading: false } );
+			} )
+			.catch( () => {
+				this.setState( { products: [], loading: false } );
+			} );
+	}
+
+	componentDidUpdate( prevProps, prevState ) {
+		if ( prevState.product !== this.state.product ) {
+			this.debouncedGetVariations();
+		}
+	}
+
+	getVariations() {
+		const { product, variationsList } = this.state;
+		if ( ! product ) {
+			this.setState( {
+				variationsList: {},
+				variationsLoading: false,
+			} );
+			return;
+		}
+		if ( ! variationsList[ product ] ) {
+			this.setState( { variationsLoading: true } );
+		}
+
+		apiFetch( {
+			path: addQueryArgs( `/wc/blocks/products/${ product }/variations`, {
+				per_page: -1,
+			} ),
+		} )
+			.then( ( variations ) => {
+				variations = variations.map( ( variation ) => ( { ...variation, parent: product } ) );
+				this.setState( ( prevState ) => ( {
+					variationsList: { ...prevState.variationsList, [ product ]: variations },
+					variationsLoading: false,
+				} ) );
 			} )
 			.catch( () => {
-				this.setState( { list: [], loading: false } );
+				this.setState( { termsLoading: false } );
 			} );
 	}
 
 	onSearch( search ) {
 		const { selected } = this.props;
 		getProducts( { selected, search } )
-			.then( ( list ) => {
-				this.setState( { list, loading: false } );
+			.then( ( products ) => {
+				this.setState( { products, loading: false } );
 			} )
 			.catch( () => {
-				this.setState( { list: [], loading: false } );
+				this.setState( { products: [], loading: false } );
+			} );
+	}
+
+	onProductSelect( item, isSelected ) {
+		return () => {
+			this.setState( {
+				product: isSelected ? 0 : item.id,
 			} );
+		};
+	}
+
+	renderItem( args ) {
+		const { item, search, depth = 0, isSelected, onSelect } = args;
+		const { product, variationsLoading } = this.state;
+		const classes = [
```

Comment:
We could use the `classNames` package here, right? I guess it would simplify the code a bit.